packages/jasmine/src/coverage_runner.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2017 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const childProcess = require('child_process');
+
+/**
+ * This is designed to collect the coverage of one target, since in nodejs
+ * and using NODE_V8_COVERAGE it may produce more than one coverage file, however bazel expects
+ * there to be only one lcov file. So this collects up the v8 coverage json's merges them and
+ * converts them to lcov for bazel to pick up later.
+ * TODO: The functionality in this script could be put into _lcov_merger then it would apply to
+ * any tool reporting coverage not just jasmine
+ */
+async function main() {
+  const result = childProcess.spawnSync(
+      process.execPath,
+      [...process.execArgv, __dirname + '/jasmine_runner.js', ...process.argv.slice(2)],
+      {stdio: 'inherit'});
+
+  const exitCode = result.status;
+
+  // 0 indicates success
+  // so if it's anything tthen exit here since bazel only collects coverage on success
+  if (exitCode !== 0) {
+    process.exit(exitCode);
+  }
+
+  if (!coverageEnabled) {
```

Comment:
Could shorten this to just
```
if (exitCode !== 0 || !coverageEnabled) {
    if (result.error) {
      console.error(result.error);
    }
   process.exit(exitCode || 1);
}
```
The `|| 1` is to cover the case of `result.status === null` if the subprocess terminates due to signal (https://nodejs.org/api/child_process.html#child_process_child_process_spawnsync_command_args_options).

Logging out result.error if it exists is helpful as well as otherwise the failure is silent.

Reply:
Although swapped to `process.exit(exitCode == null ? 1 : exitCode)` since 0 is falsey :P